packages/rocketchat-lib/server/lib/metrics.js

```diff
@@ -1,9 +1,138 @@
+RocketChat.metrics.totalLivechatMessages = new client.Gauge({'name': 'livechat_messages_total', 'help': 'total of messages in livechat rooms'});
+
+client.register.setDefaultLabels({
+	uniqueId: RocketChat.settings.get('uniqueID'),
+	siteUrl: RocketChat.settings.get('Site_Url')
+});
+
+const setPrometheusData = () => {
+	const date = new Date();
+
+	client.register.setDefaultLabels({
+		unique_id: RocketChat.settings.get('uniqueID'),
+		site_url: RocketChat.settings.get('Site_Url'),
+		version: RocketChat.Info.version
+	});
+
+	RocketChat.metrics.ddpSessions.set(Object.keys(Meteor.server.sessions).length, date);
+	RocketChat.metrics.ddpConnectedUsers.set(_.compact(_.unique(Object.values(Meteor.server.sessions).map(s => s.userId))).length, date);
+
+	const statistics = RocketChat.models.Statistics.findLast();
+	if (!statistics) {
+		return;
+	}
+
+	RocketChat.metrics.version.set({version: statistics.version}, 1, date);
+	RocketChat.metrics.migration.set(RocketChat.Migrations._getControl().version, date);
+	RocketChat.metrics.instanceCount.set(statistics.instanceCount, date);
+	RocketChat.metrics.oplogEnabled.set({enabled: statistics.oplogEnabled}, 1, date);
+
+	// User statistics
+	RocketChat.metrics.totalUsers.set(statistics.totalUsers, date);
+	RocketChat.metrics.activeUsers.set(statistics.activeUsers, date);
+	RocketChat.metrics.nonActiveUsers.set(statistics.nonActiveUsers, date);
+	RocketChat.metrics.onlineUsers.set(statistics.onlineUsers, date);
+	RocketChat.metrics.awayUsers.set(statistics.awayUsers, date);
+	RocketChat.metrics.offlineUsers.set(statistics.offlineUsers, date);
+
+	// Room statistics
+	RocketChat.metrics.totalRooms.set(statistics.totalRooms, date);
+	RocketChat.metrics.totalChannels.set(statistics.totalChannels, date);
+	RocketChat.metrics.totalPrivateGroups.set(statistics.totalPrivateGroups, date);
+	RocketChat.metrics.totalDirect.set(statistics.totalDirect, date);
+	RocketChat.metrics.totalLivechat.set(statistics.totlalLivechat, date);
+
+	// Message statistics
+	RocketChat.metrics.totalMessages.set(statistics.totalMessages, date);
+	RocketChat.metrics.totalChannelMessages.set(statistics.totalChannelMessages, date);
+	RocketChat.metrics.totalPrivateGroupMessages.set(statistics.totalPrivateGroupMessages, date);
+	RocketChat.metrics.totalDirectMessages.set(statistics.totalDirectMessages, date);
+	RocketChat.metrics.totalLivechatMessages.set(statistics.totalLivechatMessages, date);
+};
+
+const app = connect();
+
+// const compression = require('compression');
+// app.use(compression());
+
+app.use('/metrics', (req, res) => {
+	res.setHeader('Content-Type', 'text/plain');
+	res.end(RocketChat.promclient.register.metrics());
+});
+
+const server = http.createServer(app);
+
+let timer;
+RocketChat.settings.get('Prometheus_Enabled', (key, value) => {
+	if (value === true) {
+		server.listen({
+			port: 9100,
+			host: process.env.BIND_IP || '0.0.0.0'
```

Comment:
Can we re-use instance ip?  I don't think we are using bind ip anywhere

Reply:
@geekgonecrazy that is a copy from internal meteor code